schema/crdb/dbinit.sql

```diff
@@ -4064,6 +4064,55 @@ VALUES (
 ON CONFLICT (id)
 DO NOTHING;
 
+CREATE TYPE IF NOT EXISTS omicron.public.migration_state AS ENUM (
+  'pending',
+  'in_progress',
```

Comment:
Out of curiosity, what's the distinction between these two? Is it when the source stops running, that the sled agent reports a new state and we transition to `in_progress`?

Reply:
`Pending` is the initial state when the migration record is inserted into the database, *before* a sled-agent has confirmed that the migration has actually started; the state advances to `InProgress` only after a sled-agent reports back that its Propolis has actually started the migration. Initially, I didn't differentiate between these two states (see https://github.com/oxidecomputer/omicron/pull/5859/commits/631ba0f18edd9f1ad83e8d64073bcc243b9bac3f), but I thought it might be nice to be able to capture a difference between cases where one side of the migration fails before the actual migration has started, and cases where one side fails *during* the migration.

Right now, none of the code that will use this in the `instance-update` saga actually needs to distinguish this. Mostly, I thought it would be useful for debugging tools like a potential future `omdb db migration` command, but I figured we may as well be recording it.